### PR TITLE
ci: Add Django 4.1

### PR DESCRIPTION
**Pull request by adamchainz**
## Proposed changes

Add Django 4.1 to the test matrix in tox and github actions. Also simplify the tox configuration by switching `django-<V>` to `django<V>`, which allows tox to treat the whole thing as a tag which can be selected on in the `deps` section.

## Types of changes

Please check the type of change your PR introduces:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (PEP8, lint, formatting, renaming, etc)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes (build process, tests runner, etc)
- [ ] Other (please describe):

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask. We're here to
help! This is simply a reminder of what we are going to look for before merging
your code._

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

## Other information

n/a

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 deps =
-    py{36,37,38,39,310}-django-22: Django>=2.2,<2.3
-    py{36,37,38,39,310}-django-32: Django>=3.2,<3.3
-    py{38,39,310}-django-40: Django>=4.0,<4.1
-    py{38,39,310}-django-main: https://github.com/django/django/archive/main.tar.gz
+    django22: Django>=2.2,<2.3
+    django32: Django>=3.2,<3.3
+    django40: Django>=4.0,<4.1
+    django41: Django>=4.1,<4.2
+    djangomain: https://github.com/django/django/archive/main.tar.gz
```
Comment (atodorov) [inline tox.ini:18]: +1 for adding Django 4.1 but why remove the dash characters? They make things just a tiny bit easier to read.

IMO the dashes should stay. I'll leave others to chime in on this as well.

**Review by atodorov — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 1 above)
Comment (adamchainz) [reply tox.ini:18]: As I said in the PR title, tox sees names as tags separated by dashes. By putting a dash between `django` and the version number, you can't use `django<V>` as a tag, which I've used to simplify `deps` here. Putting the dash back in will require the previous format of repeating the full environment names, which seems error prone.

Comment (nijel) [reply tox.ini:18]: Is it possible to use underscore instead? To keep the readability and allow to use that as a tag?

Comment (adamchainz) [reply tox.ini:18]: I don't know, feel free to experiment and make a suggestion on github if you really care about it.

File: tox.ini [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
     coverage run manage.py test
 deps =
-    py{36,37,38,39,310}-django-22: Django>=2.2,<2.3
-    py{36,37,38,39,310}-django-32: Django>=3.2,<3.3
-    py{38,39,310}-django-40: Django>=4.0,<4.1
-    py{38,39,310}-django-main: https://github.com/django/django/archive/main.tar.gz
+    django22: Django>=2.2,<2.3
+    django32: Django>=3.2,<3.3
+    django40: Django>=4.0,<4.1
+    django41: Django>=4.1,<4.2
```
Comment (aaditya-ridecell) [inline tox.ini:17]: @adamchainz After upgrading to python 3.10 and Django 4.1, it is creating a new migration file. Did you observe this?
Attaching a sample SS:
Screenshot 2022-11-01 at 2 11 43 AM

**Comment by adamchainz** [comment]
@atodorov @nijel are you able to review now?

File: tox.ini (hunk 2 above)
Comment (aaditya-ridecell) [reply tox.ini:17]: Thank you @adamchainz!

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**